### PR TITLE
feat: Autogenerate component & prop autocompletion for Margarita components

### PR DESCRIPTION
**Pull request by afontcu**
This PR adds a first iteration of the component autocompletion & intellisense that Vetur can offer.

## What does the PR do

It yields the following result:

Captura de Pantalla 2021-01-11 a les 11 42 16

imatge

## What's missing

It doesn't add "values" or description for each prop. This needs to be done manually by adding JSDoc comments to each prop. [Here](https://github.com/CyCraft/vue-intellisense/blob/production/packages/scripts/test/helpers/BlitzForm.vue#L9) you have an example of available fields.

btw, these JSDoc comments will also be used by Storybook or any other platform to document components, so work done there will be used everywhere and will become the source of truth for props documentation. Quite neat.

### CODE REVIEW

**Comment by afontcu** [comment]
By the way, this is what files in `/vetur` look like:

Captura de Pantalla 2021-01-11 a les 12 09 15

Captura de Pantalla 2021-01-11 a les 12 09 27

File: package.json [hunk 1]
```diff
@@ -12,11 +12,12 @@
...
     "dist",
     "src",
-    "types/*.d.ts"
+    "types/*.d.ts",
+    "vetur"
   ],
   "scripts": {
-    "postinstall": "npm run tokens:build",
-    "prebuild": "npm run tokens:build",
+    "postinstall": "npm run tokens:build && npm run vetur:build",
```
Comment (afontcu) [inline package.json:19]: So that we have a better DX when developing Margarita locally, too.

**Comment by Samitier** [comment]
Amazing 🤩 

One thing not entirely related to this: do you think we should make our projects open a prompt in vscode with vetur as a "workspace recommendation"? Because it's kinda already a prerequisite to be able to code correctly (also vetur, eslint, prettier, etc).

https://tattoocoder.com/recommending-vscode-extensions-within-your-open-source-projects/

Not sure where to put it to share it between projects, though.

**Review by Samitier — APPROVED** [review]

**Comment by afontcu** [comment]
> One thing not entirely related to this: do you think we should make our projects open a prompt in vscode with vetur as a "workspace recommendation"? 

Sure! We actually have it in [bonasera](https://github.com/holaluz/bonasera/tree/master/.vscode). Not sure it is updated.

> Not sure where to put it to share it between projects, though.

Hm, it would be quite cool if we could extend recommendations, so a base file could live in npm-scripts. Let me check it out.

**Comment by edufarre** [comment]
Looks promising! Can't wait for it!

**Review by edufarre — APPROVED** [review]

**Pull request merged**